ext/MadNLPMOI/utils.jl

```diff
@@ -201,24 +201,25 @@ function eval_sparse_gradient(
     f::MOI.ScalarQuadraticFunction{T},
     x::Vector{T},
     p::Dict{Int64,T},
+    adj::T,
 )::Int where {T}
     i = 0
     for term in f.affine_terms
         if !_is_parameter(term.variable)
             i += 1
-            ∇f[i] = term.coefficient
+            ∇f[i] += term.coefficient * adj
```

Comment:
should this be `+=`? If we're evaluating non-compressed sparse grad, we wouldn't need `+=` here

Reply:
NVM, I think it is possible that there is redundant entries